openfe/setup/atom_mapping/perses_mapper.py

```diff
@@ -50,24 +51,31 @@ def __init__(self, allow_ring_breaking: bool = True,
         self.coordinate_tolerance = coordinate_tolerance
 
     def _mappings_generator(self, molA, molB):
+        # Construct Perses Mapper
         _atom_mapper = AtomMapper(
             use_positions=self.use_positions,
             coordinate_tolerance=self.coordinate_tolerance,
             allow_ring_breaking=self.allow_ring_breaking)
 
-        # Type of mapping
+        # Try generating a mapping
         try:
             _atom_mappings = _atom_mapper.get_all_mappings(
                     old_mol=molA.to_openff(), new_mol=molB.to_openff())
         except InvalidMappingException:
             _atom_mappings = []
 
+        # Catch empty mappings here
+        if (_atom_mappings is None):
+            _atom_mappings = []
+
         # Post processing
         if (self.preserve_chirality):
             [x.preserve_chirality() for x in _atom_mappings]
```

Comment:
```suggestion
        if self.preserve_chirality:
            for x in _atom_mappings:
                x.preserve_chirality()
```

1. [Don't use superfluous parens](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/superfluous-parens.html)
2. [Don't use a list comprehension if you're not assigning it](https://treyhunner.com/2019/03/abusing-and-overusing-list-comprehensions-in-python/#Loops_disguised_as_comprehensions)